src/status_im/contexts/chat/messenger/messages/view.cljs

```diff
@@ -19,57 +19,50 @@
-      [rn/keyboard-avoiding-view
-       {:style                    (style/keyboard-avoiding-container theme)
-        :keyboard-vertical-offset (- (if platform/ios? alert-banners-top-margin 0) (:bottom insets))}
-       [:<>
-        [list.view/messages-list-content props]
-        [scroll-to-bottom/button props]]
-       [messages.navigation/view props]
-       [footer props]])))
-
-(defn lazy-chat-screen
-  [chat-screen-layout-calculations-complete? *screen-loaded?*]
-  (let [screen-loaded?         (rf/sub [:shell/chat-screen-loaded?])
-        distance-from-list-top (reanimated/use-shared-value 0)
-        chat-list-scroll-y     (reanimated/use-shared-value 0)
-        props                  {:insets (safe-area/get-insets)
-                                :content-height (atom 0)
-                                :layout-height (atom 0)
-                                :distance-atom (atom 0)
-                                :distance-from-list-top distance-from-list-top
-                                :chat-list-scroll-y chat-list-scroll-y
-                                :chat-screen-layout-calculations-complete?
-                                chat-screen-layout-calculations-complete?}]
-    (when *screen-loaded?*
-      (rn/use-mount #(reset! *screen-loaded?* true)))
-    (when-not (if *screen-loaded?* @*screen-loaded?* screen-loaded?)
-      (reanimated/set-shared-value chat-screen-layout-calculations-complete? false)
-      (reanimated/set-shared-value distance-from-list-top 0)
-      (reanimated/set-shared-value chat-list-scroll-y 0))
-    (when (if *screen-loaded?* @*screen-loaded?* screen-loaded?)
-      [chat-screen props])))
+        insets                   (safe-area/get-insets)]
+    [rn/keyboard-avoiding-view
+     {:style                    (style/keyboard-avoiding-container theme)
+      :keyboard-vertical-offset (- (if platform/ios? alert-banners-top-margin 0)
+                                   (:bottom insets))}
+     [:<>
+      [list.view/messages-list-content
+       {:insets                 insets
+        :distance-from-list-top distance-from-list-top
+        :chat-list-scroll-y     chat-list-scroll-y
+        :layout-height          layout-height
+        :on-layout-done?        on-layout-done?}]
+      [scroll-to-bottom/button chat-list-scroll-y]]
+     [messages.navigation/view]
+     [footer layout-height]]))
 
 (defn chat
   []
-  (let [chat-screen-layout-calculations-complete? (reanimated/use-shared-value false)
-        jump-to-enabled?                          (ff/enabled? ::ff/shell.jump-to)
-        *screen-loaded?*                          (when-not jump-to-enabled?
-                                                    (reagent/atom false))]
-    [:<>
-     [lazy-chat-screen chat-screen-layout-calculations-complete? *screen-loaded?*]
-     [placeholder.view/view chat-screen-layout-calculations-complete?]]))
+  (let [on-layout-done?    (reagent/atom false)
+        first-render-done? (reagent/atom false)]
```

Comment:
Could the `on-layout-done?` ratom be avoided smh? Like having a value in the db for it maybe or smh else, or does that make it much slower to open? 

Asking cause it's passed through a few children, mutated "somewhere" and used in another "somewhere", so I wonder about the complexity trade-offs you mentioned, if any.

Reply:
Thanks for this comment and for noticing it!

I know this ratom is being passed, but we need to use the advantages of being a ratom and deref it in its children.

It can't be substituted with a hook (unless we use the js/requestAnimationFrame to update its value).

It could be moved to re-frame, but I think there's no need to move this local state to be a global one.

This ratom was originally called something like `chat-screen-layout-calculations-complete`, and what I did was rename it and simplify the code, but I preserved the same main logic, so it means the behavior is the same.

I can rework the logic and bring a simpler one, but I didn't want to change it because the issue solved is mostly about a redesign.

I can open a new issue with a list of code improvements we want to fix in the chat screen and then solve it. Also, the review will be easier.  wdyt?

Reply:
> It could be moved to re-frame, but I think there's no need to move this local state to be a global one.

Since it's mutated and derefed in nested children, I'd argue it doesn't/shouldn't fall under local state anymore, but I agree that this will introduce even more changes and could be should outside this PR :+1:.